CONTRIBUTING.rst

```diff
@@ -86,93 +83,111 @@ An Example
+(so as to keep their total number small). 
+
+As time passes, you make some changes to files, and you commit those changes (to
+your *local ``work`` branch*). Eventually (hopefully) you come to a stopping
+point where you have finished your project on your ``work`` branch *AND* it
+compiles *AND* it runs input files correctly *AND* it passes all tests! Perhaps
+you have found Nirvana. 
+
+Over this time, it is possible that the ``main`` branch into which you are
+proposing your pull request has advanced with other changes. In order to make
+sure your ``work`` branch remains up to date, you will want to periodically
+rebase your ``work`` branch onto the ``upstream/main`` branch.  This
+process will reapply all of the changes you have made on top of the most
+up-to-date version of the ``upstream/main`` branch.  Even if you have not been
+doing this regularly, you'll want to do it before you initiate a pull request.
+::
+
+  $ git checkout main
+  $ git pull upstream main
+  $ git push origin main
+  $ git checkout work
+  $ git rebase main
+  $ git push origin work
+
+Note: you may need to force the push of the rebased ``work`` branch to your fork
+(i.e., `git push -f origin work`).
+
+Once you've performed the final commit to your ``work`` branch it's
+time to make a pull request online and wait for our main friends to review and
+accept it.  
+
+Sometimes, your pull request will be held by the reviewer until
+further changes are made to appease the reviewer's concerns. This may be
+frustrating, but please act rationally; discuss the issues on the GitHub space
+made for your pull request, consult the `style guide
+<http://cyclus.github.com/devdoc/style_guide.html>`_, reach out on `slack
+<https://cyclus-nuclear.slack.com>`_ for further advice, and make changes to
+your ``work`` branch accordingly. The pull request will be updated with those
+changes when you push them to your fork. When you think your request is ready
+for another review, you can reopen the review yourself with the button made
+available to you.
+
+Synchronizing across multiple computers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You may wish to work on different computers over time, sometime on your laptop
+and other times on a desktop at the office. If you keep your fork's branches up
+to date (i.e., "push" your changes before you leave), only your *local* copies
+of your branches may be different when you next sit down at the other location.
+
+Let us imagine that you have been at the office, finished, and successfully
+pushed your changes to your ``origin`` repository. You are now at home and want
+to continue working a bit on your laptop. To begin, let's update our *laptop's
+local branches*
+::
+  $ git checkout work
+  $ git pull origin work
+
+This may also be a good time to ensure your ``work`` branch is up-to-date with the 
+``upstream/main`` branch
+::
+
+  $ git chekout main
+  $ git pull upstream main
+  $ git checkout work
```

Comment:
```suggestion
  $ git chekout main
  $ git pull upstream main
  $ git push origin main
  $ git checkout work
```
I think we should encourage pushing to your fork's main anytime you pull from upstream.